src/asciidocParser.ts

```diff
@@ -53,8 +50,18 @@ export class AsciidocParser {
           this.errorCollection.clear();
         }
 
-        const memoryLogger = asciidoctor.MemoryLogger.create();
-        asciidoctor.LoggerManager.setLogger(memoryLogger);
+        const memoryLogger = this.processor.MemoryLogger.create()
+        this.processor.LoggerManager.setLogger(memoryLogger)
+
+        this.registry = this.processor.Extensions.create()
+
+        this.processor.SyntaxHighlighter.for('highlight.js')
```

Comment:
I think that the highlight.js syntax highlighter "adapter" should only be registered once.
Furthermore, the correct way to register it is:

```js
highlightjsAdapter.register(this.processor)
```

So I would move the above statement in the `constructor`:

```js
constructor(private readonly filename: string, private errorCollection: vscode.DiagnosticCollection = null) {
  this.processor = asciidoctor()
  highlightjsAdapter.register(this.processor)
}
```

Reply:
Thank you for your patience. Done.